iox_query/src/plan/influxql/rewriter.rs

```diff
@@ -488,6 +488,49 @@ fn rewrite_field_list(stmt: &mut SelectStatement, schema: Arc<dyn SchemaProvider
     Ok(())
 }
 
+/// Resolve the outer-most `SELECT` projection list column names in accordance with the
+/// [original implementation]. The names are assigned to the `alias` field of the [`Field`] struct.
+///
+/// [original implementation]: https://github.com/influxdata/influxql/blob/1ba470371ec093d57a726b143fe6ccbacf1b452b/ast.go#L1651
+fn rewrite_field_list_aliases(field_list: &mut FieldList) -> Result<()> {
+    let mut column_aliases = HashMap::new();
```

Comment:
I see you clone the column names below, but with some type annotations I think you could get away w/:

```suggestion
    let mut column_aliases = HashMap::<&str, _>::new();
```

Reply:
The problem I ran into with `&str` being slices, is they outlive their references created in the closure for the HashMap, which lives in the outer scope.

Reply:
I resolved this during simplification, as I capture all the base fields names in a `Vec` first (to fix an edge case), which allows me to borrow slices from this to use in the `HashMap`.